wrangler-core/src/main/java/io/cdap/wrangler/utils/ArithmeticOperations.java

```diff
@@ -24,380 +24,462 @@
+   * Returns null if any input value is null.
    */
-  public static BigDecimal divideq(BigDecimal x, BigDecimal y) {
-    if (x == null || y == null || y.equals(BigDecimal.valueOf(0))) {
+  public static Number add(Number... nums) throws DirectiveExecutionException {
+    NumberType outputType = getOutputType(nums);
+    if (outputType == null) {
       return null;
     }
-    return x.divide(y, BigDecimal.ROUND_HALF_EVEN).stripTrailingZeros();
-  }
-
-  /**
-   * Arithmetic operation - Divide a column by another and return the remainder.
-   */
-  public static Integer divider(Integer x, Integer y) {
-    if (x == null || y == null || y == 0) {
-      return null;
+    switch (outputType) {
+      case SHORT:
+        short shortSum = 0;
+        for (Number num : nums) {
+          shortSum += num.shortValue();
+        }
+        return shortSum;
+      case INTEGER:
+        int intSum = 0;
+        for (Number num : nums) {
+          intSum += num.intValue();
+        }
+        return intSum;
+      case LONG:
+        long longSum = 0;
+        for (Number num : nums) {
+          longSum += num.longValue();
+        }
+        return longSum;
+      case FLOAT:
+        float floatSum = 0;
+        for (Number num : nums) {
+          floatSum += num.floatValue();
+        }
+        return floatSum;
+      case DOUBLE:
+        double doubleSum = 0;
+        for (Number num : nums) {
+          doubleSum += num.doubleValue();
+        }
+        return doubleSum;
+      default:
+        BigDecimal bdSum = BigDecimal.valueOf(0);
+        for (Number num : nums) {
+          bdSum = bdSum.add(new BigDecimal(num.toString()));
+        }
+        return bdSum.stripTrailingZeros();
     }
-    return x % y;
   }
 
   /**
-   * Arithmetic operation - Divide a column by another and return the remainder.
+   * Arithmetic operation - subtract a column from another column.
+   * Output type is most general of input types.
+   * Returns null if any input value is null.
    */
-  public static Double divider(Double x, Double y) {
-    if (x == null || y == null || y == 0) {
+  public static Number minus(Number x, Number y) throws DirectiveExecutionException {
+    NumberType outputType = getOutputType(x, y);
+    if (outputType == null) {
       return null;
     }
-    return x % y;
-  }
-
-  /**
-   * Arithmetic operation - Divide a column by another and return the remainder.
-   */
-  public static Float divider(Float x, Float y) {
-    if (x == null || y == null || y == 0) {
-      return null;
+    switch (outputType) {
+      case SHORT:
+        return (short) (x.shortValue() - y.shortValue());
+      case INTEGER:
+        return x.intValue() - y.intValue();
+      case LONG:
+        return x.longValue() - y.longValue();
+      case FLOAT:
+        return x.floatValue() - y.floatValue();
+      case DOUBLE:
+        return x.doubleValue() - y.doubleValue();
+      default:
+        BigDecimal bdX = new BigDecimal(x.toString());
```

Comment:
Same comment as before. No need to convert to BigDecimal if the number is already a BigDecimal.